js/components/common/react_modal.js.jsx

```diff
@@ -34,19 +37,107 @@ class CourseModal extends React.Component {
       course: [],
       sessions: [],
       courseTitle: "",
+      allDataF: [],
+      allDataS: [],
     }
   }
 
+  convertToDay(day) {
+    switch (day) {
+      case 0:
+        return "Monday"
+      case 1:
+        return "Tuesday"
+      case 2:
+        return "Wednesday"
+
+      case 3:
+        return "Thursday"
+
+      case 4:
+        return "Friday"
+
+      case 5:
+        return "Friday"
+    }
+  }
+
+  getTable(sessions) {
+    let allData = []
+    sessions.map(lecture => {
+      const occurrences = { times: [], Rooms: [] }
```

Comment:
As a stylistic thing, in general please use lowercase keys (so `rooms` instead of `Rooms`). This applies throughout these changes.